services/slam/fake/slam_test.go

```diff
@@ -34,6 +39,31 @@ func TestFakeSLAMPosition(t *testing.T) {
 	test.That(t, pInFrame2, test.ShouldResemble, pInFrame2)
 }
 
+func TestFakeSLAMGetPosition(t *testing.T) {
+	expectedComponentReference := ""
+	slamSvc := &SLAM{Name: "test", logger: golog.NewTestLogger(t)}
+
+	p, componentReference, err := slamSvc.GetPosition(context.Background(), slamSvc.Name)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, componentReference, test.ShouldEqual, expectedComponentReference)
+
+	// test.ShouldBeBetween is used here as tiny differences were observed
+	// in floating point values between M1 mac & arm64 linux which
+	// were causing tests to pass on M1 mac but fail on ci.
+
+	test.That(t, p.Point().X, test.ShouldBeBetween, -0.005666600784453, -0.005666600784454)
+	test.That(t, p.Point().Y, test.ShouldBeBetween, -6.933830e-10, -6.933831e-10)
+	test.That(t, p.Point().Z, test.ShouldBeBetween, -0.01303046, -0.01303047)
+
+	expectedOri := &spatialmath.Quaternion{Real: 0.9999999087728241, Imag: 0, Jmag: 0.0005374749356603168, Kmag: 0}
+	test.That(t, p.Orientation(), test.ShouldResemble, expectedOri)
+
+	p2, componentReference, err := slamSvc.GetPosition(context.Background(), slamSvc.Name)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, componentReference, test.ShouldEqual, expectedComponentReference)
+	test.That(t, p, test.ShouldResemble, p2)
```

Comment:
[minor] if these are string should they be equal and not just resemble?

Reply:
We can't use equal here b/c we are comparing 2 different interface types & equality is going to return false if they refer to different places in memory, even if they are semantically the same.

Making the requested change outputs:
```
=== RUN   TestFakeSLAMGetPosition
    logger.go:130: 2023-03-03T11:20:56.075-0500 DEBUG   fake/data_loader.go:180 Reading /Users/nicksanford/rdk/.artifact/data/slam/example_cartographer_outputs/viam-office-02-22-1/position_new/position_0.json
    slam_test.go:62: Both the actual and expected values render equally ('&{0.9999999087728241 0 0.0005374749356603168 0}') and their types are the same. Try using ShouldResemble instead.
--- FAIL: TestFakeSLAMGetPosition (0.13s)
FAIL
exit status 1
FAIL    go.viam.com/rdk/services/slam/fake      0.553s
zsh returned exit code 1

```